### DISCUSSION TITLE
How can I connect html to react?

### DISCUSSION THREAD

**Discussion by GroovyMarket** [community/community#123190] [Codespaces] [answered]
### Select Topic Area

Question

### Body

Hello, everyone!
I wanna know how to connect html to react.
Especially include index.html.

**Comment by techpond31** [comment, marked as answer]
👋 Hello! To include your React app in an HTML file, you typically use a bundler like Webpack, Parcel, or Rollup to produce a compiled JavaScript file that handles rendering your React components. Here's an example of how you can set up a minimal HTML file and a simple React app.

1. Create a new directory for your project and navigate into it:

```
mkdir react-example && cd react-example
```

2. Initialize the project with `npm`:

```
npm init -y
```

3. Install required packages:

```
npm install --save react react-dom
npm install --save-dev parcel-bundler
```

4. Create the following files:

index.html:

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>React Example</title>
  </head>
  <body>
    <div id="root"></div>
    <script src="./src/index.js"></script>
  </body>
</html>
```

src/index.js:

```javascript
import React from 'react';
import ReactDOM from 'react-dom';

const App = () => {
  return (
    <div>
      <h1>Hello, world!</h1>
    </div>
  );
};

ReactDOM.render(<App />, document.getElementById('root'));
```

5. Add build and serve scripts to package.json:

```json
"scripts": {
  "start": "parcel index.html",
  "build": "parcel build index.html"
}
```

6. Serve the project:

```
npm start
```

7. Build the project:

```
npm run build
```

Now, you should have a simple React app served locally. The `index.html` file contains a placeholder `<div id="root"></div>`, and the compiled JavaScript fetches and renders the React app inside that `div`. This is a basic example of how you can incorporate React into an HTML file.
